README.md

```diff
@@ -840,6 +841,95 @@ public class Handler : IRequiresAsyncInitialization
+
+```cs
+using Microsoft.Data.SqlClient;
+using Microsoft.EntityFrameworkCore;
+using StrongInject;
+using System;
+using System.Threading.Tasks;
+
+[Register(typeof(SomeViewModel))]
+[Register(typeof(SomeDbContext))]
+public partial class Container : IContainer<SomeViewModel>
+{
+    [Factory]
+    public static SqlConnection CreateSqlConnection()
+    {
+        throw new NotImplementedException();
+    }
+}
+
+public class SomeViewModel
+{
+    private readonly Func<Owned<SomeDbContext>> dbContextFactory;
+
+    public SomeViewModel(Func<Owned<SomeDbContext>> dbContextFactory)
+    {
+        this.dbContextFactory = dbContextFactory;
+    }
+
+    public async Task SaveAsync(string newName)
+    {
+        // Using Func<Owned<SomeDbContext>> instead of Func<SomeDbContext> ensures that the context will not be
+        // disposed while this method is still using it, even if SomeViewModel is released back to the container.
+
+        // Using Func<Owned<SomeDbContext>> instead of Owned<SomeDbContext> gives the ability for SaveAsync to
+        // get a new SomeDbContext instance that is not being used anywhere else, so overlapping calls to SaveAsync
+        // are no problem.
+
+        // By modeling the dependency this way, SomeViewModel remains agnostic about whether SomeDbContext is
+        // outliving it by being pooled and recycled (not shown here). Both SomeViewModel and SomeDbContext are
+        // even agnostic about whether the SqlConnection is being pooled and recycled.
+
+        // Each component is agnostic about how its dependencies are created and cleaned up. This allows decisions
+        // to be changed in the configuration of the components without breaking them.
+        // This is naturally the case already when injecting a dependency directly the way SqlConnection is injected
+        // into SomeDbContext; what’s special here is showing how to achieve the same decoupling when the same
+        // SomeViewModel needs to temporarily use any number of SomeDbContext instances over its lifetime.
+
+        using var dbContext = dbContextFactory.Invoke();
+
+        dbContext.Value.Set<SomeEntity>().Add(new SomeEntity { Name = newName });
+        await dbContext.Value.SaveChangesAsync();
+    }
+}
+
+public class SomeDbContext : DbContext
+{
+    public SomeDbContext(SqlConnection connection)
+        : base(new DbContextOptionsBuilder().UseSqlServer(connection).Options)
+    {
+    }
+}
+
+public class SomeEntity
+{
+    public string Name { get; set; }
+}
+```
+
+> ℹ **Coming from another container?**
+>
+> Injecting a delegate such as `Func<…, Owned<T>>` is a concept which maps 1:1 with a concept other containers have of auto-implementing an interface you define, such as `IFactory<T>` with `T Resolve(…)` and `void Release(T)` methods. For example, Castle Windsor calls this feature _typed factories_.
```

Comment:
I don't think Autofac has this - it has an Owned type instead, much like StrongInject. Perhaps focus this specifically on Castle Windsor?